### ISSUE TITLE
Comments not loading, even though not rate limited

### ISSUE THREAD

**Issue by FireMasterK** [TeamNewPipe/NewPipeExtractor#688]
Comments do not work in NewPipe but work on the Mobile website.

Edit: Retrying a few times fixes this, but should still be investigated!

## Exception
* __User Action:__ requested comments
* __Request:__ Start loading: https://www.youtube.com/watch?v=6MtsHx1-YDg
* __Content Country:__ US
* __Content Language:__ en
* __App Language:__ en
* __Service:__ YouTube
* __Version:__ 0.21.6
* __OS:__ Linux Android 10 - 29
<details><summary><b>Crash log </b></summary><p>

```
org.schabi.newpipe.extractor.exceptions.ParsingException: Could not parse json data for comments
	at org.schabi.newpipe.extractor.services.youtube.extractors.YoutubeCommentsExtractor.getPage(YoutubeCommentsExtractor.java:100)
	at org.schabi.newpipe.extractor.services.youtube.extractors.YoutubeCommentsExtractor.getInitialPage(YoutubeCommentsExtractor.java:55)
	at org.schabi.newpipe.extractor.utils.ExtractorHelper.getItemsPageOrLogError(ExtractorHelper.java:19)
	at org.schabi.newpipe.extractor.comments.CommentsInfo.getInfo(CommentsInfo.java:40)
	at org.schabi.newpipe.extractor.comments.CommentsInfo.getInfo(CommentsInfo.java:25)
	at org.schabi.newpipe.util.ExtractorHelper.lambda$getCommentsInfo$7(ExtractorHelper.java:155)
	at org.schabi.newpipe.util.-$$Lambda$ExtractorHelper$60N_-UL7E5eaxFaFO1bZZmnfwM8.call(Unknown Source:4)
	at io.reactivex.rxjava3.internal.operators.single.SingleFromCallable.subscribeActual(SingleFromCallable.java:43)
	at io.reactivex.rxjava3.core.Single.subscribe(Single.java:4813)
	at io.reactivex.rxjava3.internal.operators.single.SingleDoOnSuccess.subscribeActual(SingleDoOnSuccess.java:35)
	at io.reactivex.rxjava3.core.Single.subscribe(Single.java:4813)
	at io.reactivex.rxjava3.internal.operators.maybe.MaybeFromSingle.subscribeActual(MaybeFromSingle.java:41)
	at io.reactivex.rxjava3.core.Maybe.subscribe(Maybe.java:5330)
	at io.reactivex.rxjava3.internal.operators.maybe.MaybeConcatArray$ConcatMaybeObserver.drain(MaybeConcatArray.java:153)
	at io.reactivex.rxjava3.internal.operators.maybe.MaybeConcatArray$ConcatMaybeObserver.request(MaybeConcatArray.java:78)
	at io.reactivex.rxjava3.internal.operators.flowable.FlowableElementAtMaybe$ElementAtSubscriber.onSubscribe(FlowableElementAtMaybe.java:66)
	at io.reactivex.rxjava3.internal.operators.maybe.MaybeConcatArray.subscribeActual(MaybeConcatArray.java:42)
	at io.reactivex.rxjava3.core.Flowable.subscribe(Flowable.java:15753)
	at io.reactivex.rxjava3.internal.operators.flowable.FlowableElementAtMaybe.subscribeActual(FlowableElementAtMaybe.java:36)
	at io.reactivex.rxjava3.core.Maybe.subscribe(Maybe.java:5330)
	at io.reactivex.rxjava3.internal.operators.maybe.MaybeToSingle.subscribeActual(MaybeToSingle.java:46)
	at io.reactivex.rxjava3.core.Single.subscribe(Single.java:4813)
	at io.reactivex.rxjava3.internal.operators.single.SingleSubscribeOn$SubscribeOnObserver.run(SingleSubscribeOn.java:89)
	at io.reactivex.rxjava3.core.Scheduler$DisposeTask.run(Scheduler.java:614)
	at io.reactivex.rxjava3.internal.schedulers.ScheduledRunnable.run(ScheduledRunnable.java:65)
	at io.reactivex.rxjava3.internal.schedulers.ScheduledRunnable.call(ScheduledRunnable.java:56)
	at java.util.concurrent.FutureTask.run(FutureTask.java:266)
	at java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.run(ScheduledThreadPoolExecutor.java:301)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1167)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:641)
	at java.lang.Thread.run(Thread.java:919)
Caused by: com.grack.nanojson.JsonParserException: Unexpected character: < on line 1, char 1
	at com.grack.nanojson.JsonTokener.createParseException(Unknown Source:44)
	at com.grack.nanojson.JsonTokener.advanceToToken(Unknown Source:118)
	at com.grack.nanojson.JsonParser.advanceToken(Unknown Source:12)
	at com.grack.nanojson.JsonParser.parse(Unknown Source:1)
	at com.grack.nanojson.JsonParser$JsonParserContext.from(Unknown Source:19)
	at org.schabi.newpipe.extractor.services.youtube.extractors.YoutubeCommentsExtractor.getPage(YoutubeCommentsExtractor.java:98)
	... 30 more

```
</details>
<hr>

**Comment by litetex** [comment]
Found the problem and described it in https://github.com/TeamNewPipe/NewPipe/issues/6744

**Comment by FireMasterK** [comment]
Fixed at https://github.com/FireMasterK/NewPipeExtractor/tree/youtubei-api.

**Comment by litetex** [comment]
@FireMasterK 
Can you create a PR for that?

**Comment by FireMasterK** [comment]
> @FireMasterK
> Can you create a PR for that?

@TiA4f8R said he/they would merge it in #604.

**Comment by Stypox** [comment]
> Fixed at https://github.com/FireMasterK/NewPipeExtractor/tree/youtubei-api.

@FireMasterK that fork works flawlessly, except for the `isHeartedByUploader` function which always returns `false`

> @TiA4f8R said he/they would merge it in #604.

If possible I'd merge it separately, so that we can get it earlier, would it be possible? Or does your PR depend on #604?

**Comment by FireMasterK** [comment]
> @FireMasterK that fork works flawlessly, except for the `isHeartedByUploader` function which always returns `false`

 I'll check this out in one or two days, I'm currently on a vacation.

> > @TiA4f8R said he/they would merge it in #604.
> 
> If possible I'd merge it separately, so that we can get it earlier, would it be possible? Or does your PR depend on #604?

It's dependant on #604, since it uses the new youtubei api which is used on the desktop website, and hence uses the helper functions.

**Comment by FireMasterK** [comment]
Looks like @TiA4f8R has fixed it already 🚀 
https://github.com/TeamNewPipe/NewPipeExtractor/pull/604/commits/f3e4c9d689d72f1c800acde289a6c9159377010a#diff-6b5235d764fb84b15b4c3e75e5e9b1d2c483da0187025c94847ebc9860ab31b4L179

**Comment by litetex** [comment]
Fixed by #604
